core/src/main/java/io/stargate/core/activator/BaseActivator.java

```diff
@@ -97,15 +100,30 @@ String constructDependenciesFilter() {
     return builder.toString();
   }
 
+  /**
+   * It is calling the user-provided {@link this#stopService()} if the service was started
+   * successfully. If there {@link this#targetServiceClass} is present, and there was an OSGi
+   * service registration it will deregister service callint {@link
+   * BundleContext#ungetService(ServiceReference)}.
+   */
   @Override
   public synchronized void stop(BundleContext context) throws Exception {
     if (started) {
       logger.info("Stopping {}", activatorName);
       stopService();
+      deregisterService();
     }
     tracker.close();
   }
 
+  private void deregisterService() {
+    if (targetServiceRegistration != null) {
+      ServiceReference<?> reference = targetServiceRegistration.getReference();
+      logger.info("Unget service {} from {}", reference.getBundle(), activatorName);
+      context.ungetService(reference);
```

Comment:
Should we also unget the dependencies?

Reply:
No, they should be automatically `unget` by the `ServiceTracker`